packages/react-reconciler/src/__tests__/ReactUpdaters-test.internal.js

```diff
@@ -0,0 +1,603 @@
+    // and idle updates are slightly higher priority than offscreen work.
+    // So it takes two render passes to finish it.
+    // The onCommit hook is called even after the no-op bailout update.
+    expect(Scheduler).toHaveYielded([
+      'App',
+      'onCommitRoot',
+      'App:mount',
+
+      'Child',
+      'onCommitRoot',
+      'Child:mount',
+
+      'onCommitRoot',
+
+      'Child',
+      'onCommitRoot',
+      'Child:update',
+    ]);
+    // Initial render
+    expect(allSchedulerTypes).toHaveLength(4);
+    expect(allSchedulerTags[0]).toHaveLength(1);
+    expect(allSchedulerTags[0]).toContain(HostRoot);
+    // Offscreen update
+    expect(allSchedulerTypes[1]).toHaveLength(0);
+    // Child passive effect
+    expect(allSchedulerTypes[2]).toHaveLength(1);
+    expect(allSchedulerTypes[2]).toContain(Child);
+    // Offscreen update
+    expect(allSchedulerTypes[3]).toHaveLength(0);
+  });
+
+  // @gate experimental
+  it('should cover error handling', async () => {
+    let triggerError = null;
+
+    const Parent = () => {
+      const [shouldError, setShouldError] = React.useState(false);
+      triggerError = () => setShouldError(true);
+      return shouldError ? (
+        <ErrorBoundary>
+          <BrokenRender />
+        </ErrorBoundary>
+      ) : (
+        <ErrorBoundary>
+          <Yield value="initial" />
+        </ErrorBoundary>
+      );
+    };
+    class ErrorBoundary extends React.Component {
+      state = {error: null};
+      componentDidCatch(error) {
+        this.setState({error});
+      }
+      render() {
+        if (this.state.error) {
+          return <Yield value="error" />;
+        }
+        return this.props.children;
+      }
+    }
+    const Yield = ({value}) => {
+      Scheduler.unstable_yieldValue(value);
+      return null;
+    };
+    const BrokenRender = () => {
+      throw new Error('Hello');
+    };
+
+    const root = ReactDOM.unstable_createRoot(document.createElement('div'));
+    await ReactTestUtils.act(async () => {
+      root.render(<Parent shouldError={false} />);
+    });
+    expect(Scheduler).toHaveYielded(['initial', 'onCommitRoot']);
+    expect(triggerError).not.toBeNull();
+
+    allSchedulerTypes.splice(0);
+    onCommitRootShouldYield = true;
+
+    await ReactTestUtils.act(async () => {
+      triggerError();
+    });
+    expect(Scheduler).toHaveYielded(['onCommitRoot', 'error', 'onCommitRoot']);
+    expect(allSchedulerTypes).toHaveLength(2);
+    expect(allSchedulerTypes[0]).toHaveLength(1);
+    expect(allSchedulerTypes[0]).toContain(Parent);
+    expect(allSchedulerTypes[1]).toHaveLength(1);
+    expect(allSchedulerTypes[1]).toContain(ErrorBoundary);
+
+    // Verify no outstanding flushes
+    Scheduler.unstable_flushAll();
+  });
+
+  // @gate experimental
+  it('should distinguish between updaters in the case of interleaved work', async () => {
+    let triggerLowPriorityUpdate = null;
+    let triggerSyncPriorityUpdate = null;
```

Comment:
nit: Match the name of the component? 

```suggestion
    let triggerHighPriorityUpdate = null;
```

Or maybe better rename the component to `SyncPriorityUpdate` since we call `triggerSyncPriorityUpdate` from `flushSync`.